### ISSUE TITLE
undefined passes enum test

### ISSUE THREAD

**Issue by Anahkiasen** [flatiron/revalidator#111]
I have this:

``` js
revalidator.validate({foo: undefined}, {
    properties: {
        foo: {
            enum: ['foo', 'bar'],
            type: 'string',
        },
    },
});
```

Which doesn't generate any error, surely `undefined` is not `['foo', 'bar']` nor is it a string, shouldn't this throw an error?

**Comment by cwp** [comment]
Only if you have `required: true` in the schema for foo.

**Comment by Anahkiasen** [comment]
I tried with required but IIRC it passed as well, I had to convert the rule to a `pattern: '(foo|bar)'` currently. Will double check.
